### ISSUE TITLE
Cosign copy multi arch/platform support

### ISSUE THREAD

**Issue by shreelola** [sigstore/cosign#2748]
**Description**

Currently, when we build and sign a multi-platform container image, cosign creates one top-level signature and other platform-specific signatures. But when we use `cosign copy` with the specific platform, we cannot copy the signature for the specific platform, which leads to the sign verification error. 
Please add a command to copy only specific platform.

**Comment by Gabrielopesantos** [comment]
Hello,
I would like to work on this issue. Is there anything I should know before starting (Besides the [contributing guidelines](https://github.com/sigstore/cosign/blob/main/CONTRIBUTING.md))? Thank you

**Comment by Gabrielopesantos** [comment]
Hello @cpanato,
Sorry for the long delay since my last message. I finally had some time and patience to understand why I couldn't replicate the above-described issue locally (was following the scaffolding "get started" documentation and had to use an earlier version of cosign). 
Now that I managed to replicate the issue locally I'm not sure if I'm understanding what is being requested/asked. I managed to sign a multi-arch/platform image and copy all signatures into another registry (with `--sig-only` it seemed to leave out some signatures). Just to confirm, is what's being asked the possibility of copying a specific arch/platform?
